internal/api/v1/stage.go

```diff
@@ -120,6 +121,11 @@ func (hc ApplicationsController) Stage(w http.ResponseWriter, r *http.Request) A
 		Instances: instances,
 	}
 
+	err = application.Unstage(name, org, uid)
```

Comment:
I can't remember does this also remove the workload? So would this interfere with zero-downtime?

Reply:
`Unstage` is new. It removes the PipelineRuns. The workload did not look affected in my tests, watching the pod stati.

Reply:
I was thinking that the old pipeline run (and staging pods) should be deleted only when the new staging succeeds. The reason is that, if the new staging process fails, the old version of the app will keep running and the user may decide to stick with that version for a while (e.g. until they fix the new version which fails). In that case, a user could still ask for the staging logs of the running version. This line would have deleted those and they won't be available anymore.

In other words, only get rid of old staging when you got a new successful one.

Reply:
The new code should be movable to a place where we know that the new staging has succeeded or not.
That said, where (in the server) do we know when staging of an app has succeeded ?

Reply:
Right now I think it's on the client only. I created this spike to see if there is a way to achieve this on the server side: https://github.com/epinio/epinio/pull/443#discussion_r644510047